Towny/src/main/java/com/palmergames/bukkit/towny/db/TownyDatabaseHandler.java

```diff
@@ -117,12 +117,14 @@ protected TownyDatabaseHandler(Towny plugin, TownyUniverse universe) {
 			}
 		
 		/*
-		 * Start our Async queue for pushing data to the flatfile database.
+		 * Start our async queue for pushing data to the database.
 		 */
 		task = plugin.getScheduler().runAsyncRepeating(() -> {
-			while (!this.queryQueue.isEmpty()) {
-				Runnable operation = this.queryQueue.poll();
-				operation.run();
+			synchronized(queryQueue) {
+				while (!this.queryQueue.isEmpty()) {
+					Runnable operation = this.queryQueue.poll();
+					operation.run();
+				}
```

Comment:
Using this synchronized, is this how we get around the issue that required the SQL source to use the isPolling variable?

Reply:
That's the idea yes